### DISCUSSION TITLE
Formula: Achieving desired rigidbody height factoring in active gravity, by applying calculated force

### DISCUSSION THREAD

**Discussion by Mangonels** [godot-jolt/godot-jolt#927] [Questions]
This is something I always struggled with on any game engine, since physics are inbuilt. Jolt physics is no exception.

I'm looking for a formula, let's think of it as a gdscript function, that would check the scene's gravity, and receive a height by parameter. The function is meant to return the necessary applied force (meant to be input into a `RigidBody3D.apply_central_force()`) in order to reach this height.

How would you (if even posible) code this gdscript func? I'm looking for ideas at the very least, if there is no precise formula posible.

You can imagine I would be attempting to get a rigidbody on my scene to reach a certain (variable) height (with a fair amount of precision), but I don't know what the force necessary to be applied should be to reach it. Of course, asuming this RigidBody3D will only receive a single force at any time, and that is, the force to reach this height.

All other potential forces manually applied to it (except for the default gravity) are irrelevant/should be refrained for this experiment to conclude as "working for intended purposes".

**Comment by mihe** [comment]
I honestly don't know. Despite having worked on this extension for a while now I am not really a physics programmer, nor was I ever very good at physics/math in general.

Either way, this isn't really a Jolt question per se, so I would suggest you reach out to any of the more general [Godot communities](https://godotengine.org/community/) instead and see if they can help you.

I can also recommend trying out any of the more popular LLM chat bots, like [claude.ai](https://claude.ai/), which I've had some success with when it comes to figuring out more general software/math problems like this (as opposed to more Godot-specific one).

**Comment by Mangonels** [comment]
The only approach I can think of would be actually mapping out height to force, by examining the height reached from a different force applied, based on the simulation itself. Then mapping out how gravity would affect, and coming out with some sort of wacky formula based on that data hahaha

but I'll keep asking and I'll check where you suggested, thanks.
